hack/generator/pkg/astmodel/structtype.go

```diff
@@ -132,5 +133,19 @@ func (structType *StructType) RelatedDefinitions(ref PackageReference, namehint
 		}
 	}
 
+	for name, function := range structType.functions {
+		funcDef := &FunctionDefinition{
+			Function: function,
+			DefinitionName: DefinitionName{ref, name, &namehint},
+			Receiver: &StructReference{DefinitionName{ref, namehint, nil}, false}} // TODO: This false right?
```

Comment:
I created an item for this and will leave it as is in this PR

Reply:
For the StructReference it's not just cleanup it's a correctness issue (or will be in future when we actually use the referenced-resource thing).

Reply:
We need `RelatedDefintions` on _both_ `Definition` and `Type` because it's often (usually?) the Type that needs to generate the definitions - this is true at least for Enums; I've made the requisite change as a part of #101